packages/connection-model/lib/model.js

```diff
@@ -805,11 +817,42 @@ Connection = AmpersandModel.extend({
     }
 
     if (attrs.authStrategy === 'KERBEROS') {
-      if (!attrs.kerberosServiceName) {
-        attrs.kerberosServiceName = KERBEROS_SERVICE_NAME_DEFAULT;
-      }
+      const authProperties = attrs.authMechanismProperties || {};
+      this.kerberosServiceName = defaultTo(
+        attrs.kerberosServiceName,
+        defaultTo(attrs.gssapiServiceName, authProperties.SERVICE_NAME)
+      );
+
+      this.kerberosServiceRealm = defaultTo(
+        attrs.kerberosServiceRealm,
+        defaultTo(attrs.gssapiServiceRealm, authProperties.SERVICE_REALM)
+      );
 
-      this.kerberosServiceName = attrs.kerberosServiceName;
+      this.kerberosCanonicalizeHostname = defaultTo(
+        attrs.kerberosCanonicalizeHostname,
+        defaultTo(
+          attrs.gssapiCanonicalizeHostName,
+          defaultTo(
+            attrs.gssapicanonicalizehostname,
+            defaultTo(authProperties.CANONICALIZE_HOST_NAME, authProperties.gssapiCanonicalizeHostName)
+          )
+        )
+      );
+
+      delete attrs.gssapiServiceName;
```

Comment:
Why do we need to modify the input?

Reply:
I tried it without that but it doesn't work reliably - so for example the `gssapiServiceName` is then still set on the model. I saw it being done here: https://ampersandjs.com/docs/#ampersand-state-parse.